src/NetMQ/NetMQPoller.cs

```diff
@@ -112,21 +113,47 @@ protected override void QueueTask(Task task)
             m_tasksQueue.Enqueue(task);
         }
 
-        public void Run([NotNull] Action action)
+        public Task Run([NotNull] Action action)
         {
+            Task t;
+
             if (!IsRunning || CanExecuteTaskInline)
+            {
                 action();
+
+                t = FromResult<object>(null);
+            }
             else
-                new Task(action).Start(this);
+            {
+                t = new Task(action);
+                t.Start(this);
```

Comment:
`Task.Factory.StartNew` is preferred over `new Task(...).Start` ([article](https://devblogs.microsoft.com/pfxteam/task-factory-startnew-vs-new-task-start/)).

Reply:
ahh.... yes, I was wishing ```Thread.Start()``` was fluent.

Reply:
Even if `Thread.Start` did exist, it would be too heavyweight. We don't need or want to start a new thread for this background work. We need this task to run asynchronously on the poller's thread, which is achieved by passing `this` as the `Task`'s `TaskScheduler`.

Reply:
Sry, I said `Thread.Start `but I meant `Task.Start()`